src/main/java/com/udojava/evalex/Expression.java

```diff
@@ -1272,6 +1273,11 @@ private void shuntOperators(List<Token> outputQueue, Stack<Token> stack, LazyOpe
 	 * @return The result of the expression. Trailing zeros are stripped.
 	 */
 	public BigDecimal eval() {
+		try {
+			evalValidation();
+		} catch (Exception e) {
+			throw new ExpressionException(e.getMessage());
```

Comment:
That is evil, that will remove the original stacktrace. Please *never* do this, because it makes debugging *so much* harder.

Reply:
ok, let me think for a moment